app/controllers/comments_controller.rb

```diff
@@ -65,7 +66,12 @@ def set_comment
     end
 
     def is_an_authorized_user
-     @photo = Photo.find(params.fetch(:comment).fetch(:photo_id))
+     if @comment == nil
+      @photo = Photo.find(params.fetch(:comment).fetch(:photo_id))
+     else
+      params[:comment] = Comment.find(params[:id])
+      @photo = params[:comment].photo
+     end
       if current_user != @photo.owner && @photo.owner.private? && !current_user.leaders.include?(@photo.owner)
         redirect_back fallback_location: root_url, alert: "Not authorized"
       end
```

Comment:
You may want to remove this logic and replace it with a pundit policy ; however, it is working